### PR TITLE
Add test for js-string-builtins

### PR DESCRIPTION
**Pull request by eqrion**
This commit adds a basic suite of tests for the js-string-builtins.

This is done by defining a polyfill module matching the overview, and then comparing the host provided builtins against the polyfill on representative inputs.

cc'ing @ajklein, as we discussed tests. Do you (or someone else) want to review this? Otherwise I can just merge it.

### CODE REVIEW

**Comment by ajklein** [comment]
Happy to review this, will aim to get to it by end-of-day today.

File: test/js-api/js-string/basic.tentative.any.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  "ab",
+  "hello, world",
+  "\n",
+  "☺",
+  "☺☺",
+  String.fromCodePoint(0x10000, 0x10001)
+];
+const testCharCodes = [1, 2, 3, 10, 0x7f, 0xff, 0xfffe, 0xffff];
+const testCodePoints = [1, 2, 3, 10, 0x7f, 0xff, 0xfffe, 0xffff, 0x10000, 0x10001];
+const testExternRefValues = [
```
Comment (ajklein) [inline test/js-api/js-string/basic.tentative.any.js:191]: Is there a reason you don't test the exceptional cases here against the whole API?

**Review by ajklein — APPROVED** [review]
This mostly looks good to me, one question below. I'm also happy to land this as-is as I like the general approach, and we can easily add more as we go.

File: test/js-api/js-string/basic.tentative.any.js (hunk 1 above)
Comment (eqrion) [reply test/js-api/js-string/basic.tentative.any.js:191]: No good reason, just didn't think to do that. I'll see if I can work that in.

**Pull request merged**